atomate/vasp/workflows/base/neb.py

```diff
@@ -0,0 +1,359 @@
+    uis_ep = _get_incar(mode, user_incar_settings)[1]
+    uis_neb = _get_incar(mode, user_incar_settings)[2]
+    neb_round = len(uis_neb)
+    cust_args = cust_args or {}
+
+    # Get relaxation fireworks.
+    if is_optimized:  # Start from endpoints
+        endpoints = get_endpoints_from_index(structure, path_sites)
+        endpoints_dict = [e.as_dict() for e in endpoints]
+        spec = _update_spec_from_inputs(spec, structure=structure.as_dict(),
+                                        path_sites=path_sites, wf_name=wf_name,
+                                        endpoints=endpoints_dict)
+
+        # Get mpi vasp command
+        vasp_cmd = _get_command(spec, "std")
+        gamma_vasp_cmd = _get_command(spec, "gam")
+
+        # Get neb fireworks.
+        neb_fws = []
+        for n in range(neb_round):
+            fw = NEBFW(spec=spec, neb_label=str(n + 1), from_images=False,
+                       user_incar_settings=uis_neb[n], vasp_cmd=vasp_cmd,
+                       gamma_vasp_cmd=gamma_vasp_cmd, cust_args=cust_args)
+            neb_fws.append(fw)
+
+        # Get relax fireworks
+        rlx_fws = [NEBRelaxationFW(spec=spec, label=i, vasp_input_set=None,
+                                   user_incar_settings=uis_ep, vasp_cmd=vasp_cmd,
+                                   gamma_vasp_cmd=gamma_vasp_cmd,
+                                   cust_args=cust_args) for i in ["ep0", "ep1"]]
+
+        links = {rlx_fws[0]: [neb_fws[0]],
+                 rlx_fws[1]: [neb_fws[0]]}
+    else:  # Start from perfect structure
+        spec = _update_spec_from_inputs(spec, structure=structure.as_dict(), wf_name=wf_name,
+                                        init=structure.as_dict(), path_sites=path_sites)
+        vasp_cmd = _get_command(spec, "std")
+        gamma_vasp_cmd = _get_command(spec, "gam")
+
+        # Get neb fireworks.
+        neb_fws = []
+        for n in range(neb_round):
+            fw = NEBFW(spec=spec, neb_label=str(n + 1), from_images=False,
+                       user_incar_settings=uis_neb[n], vasp_cmd=vasp_cmd,
+                       gamma_vasp_cmd=gamma_vasp_cmd, cust_args=cust_args)
+            neb_fws.append(fw)
+
+        rlx_fws = [NEBRelaxationFW(spec=spec, label=label, vasp_input_set=None,
+                                   user_incar_settings=incar)
+                   for label, incar in zip(["init", "ep0", "ep1"], [uis_init, uis_ep, uis_ep])]
+
+        links = {rlx_fws[0]: [rlx_fws[1], rlx_fws[2]],
+                 rlx_fws[1]: [neb_fws[0]],
+                 rlx_fws[2]: [neb_fws[0]]}
+
+    # Append Firework links.
+    fws = rlx_fws + neb_fws
+    if neb_round > 1:
+        for r in range(1, neb_round):
+            links[neb_fws[r - 1]] = [neb_fws[r]]
+    workflow = Workflow(fws, links_dict=links, name=wf_name)
+    return workflow
+
+
+def get_wf_neb_from_endpoints(user_incar_settings, parent, endpoints=None, is_optimized=True,
```

Comment:
"parent" refers to as the perfect crystal structure (no impurities/defects) of interest. We completely agree that the NEB workflow can always start with two given endpoint structures. However, it will be used to label a group of NEB calculation (different paths) within the same parent structure in the future, which is a helpful tag in database insertion and query.